text/0255-sensitive-data-redaction.md

```diff
@@ -0,0 +1,397 @@
+
+The `<SENSITIVY_DETAILS>` should look like the following:
+
+```
+{
+  `<REDACTION_PROFILE>` => `<REDACTION_CONFIGURATION>`,
+  ...
+}
+```
+
+There may be multiple redaction profiles (like `DEFAULT`, `STRICT`, etc.), and
+for each one a configuration (as outlined above) may be applied.
+
+any sensitive data.
+
+## Internal details
+
+### Redactor
+
+To accomplish redaction the SDK needs a component (similar to a sampler) that inspects attributes when they are set and applies the required redactions:
+
+- `Redactor::setup(profile)` will setup the redactor with the given profile,
+  maybe a constructor? depends on the language
+- `Redactor::redact(value, <sensitivityDetails>)` will return the redacted value
+  using the provided profile
+
+The redactor will also have all the methods to apply predefined redactions
+(`REDACT_ALL_URL_VALUES`, `REDACT_INSECURE_URL_PARAM_VALUES`, `DROP`, etc.).
+If a method is not implemented (either by the SDK or by the end-user choosing
+one that does not exist), it will default to apply `DROP` to avoid leakage of
+
+## Additional context
+
+Treating sensitive data properly is a very complex multi-dimensional topic.
+Below you will find some additional context on that subject.
+
+### Types of sensitive data
+
+There are different kinds of “sensitivity” that may apply to data. The ones most
+relevant in this proposal are “security” and “privacy”. They may overlap but we
+distinguish them as follows:
+
+- security-relevant sensitive data: any information that when exposed
+  [weakens the overall security of the device/system](https://en.wikipedia.org/wiki/Vulnerability_(computing)).
+- privacy-relevant sensitive data: any information that when exposed
+  [can adversely affect the privacy or welfare of an individual](https://en.wikipedia.org/wiki/Information_sensitivity).
+
+Note, that there are other kinds of sensitive data (business information,
+classified information), which are not covered extensively by this proposal.
+
+### Level of Sensitivity
+
+The level of sensitivity of an information can also be different and that
+sensitivity can be contextual, e.g.
+
+- The password of a user without privileges is less sensitive than the password
+  of an administrator
+- The "client IP" in a server-to-server communication is less sensitive than the
+  "client IP" in an client-to-server communication, where the client can be
+  linked to a human.
+- API tokens of a demo system are less sensitive than API tokens for a production
+  system
+- The license plate of an individual’s car is less sensitive than their social
+  security number
+- The full name of a user in a social network is less sensitive than the full
+  name of a user in a medical research database
+
+Depending on the sensitivity data an end-user of an observability system may
```

Comment:
I'm not sure we need to be defining the level of sensitivity for users.  For example, a system able to be Privacy compliant could likely handle wipeout rules and protection for all of these.  OpenTelemetry should not enforce level of sensitivity, but *allow our users* to interact with their own definitions using our system.

I'd cut this section or call it out as something for our users to define, not us.

Reply:
This is already a kind of "appendix" section, I can make it clearer from the document structure.